src/Frontend/Page/Idea.hs

```diff
@@ -362,10 +362,14 @@ instance FormPage JudgeIdea where
 -- on the bright side, it makes shorter uri paths possible.)
 viewIdea :: (ActionPersist m, MonadError ActionExcept m, ActionUserHandler m)
     => AUID Idea -> m (Frame ViewIdea)
-viewIdea ideaId = makeFrame =<< (do
+viewIdea ideaId = viewIdea' ideaId >>= makeFrame
+
+viewIdea' :: (ActionPersist m, MonadError ActionExcept m, ActionUserHandler m)
```

Comment:
What do you tihink about renaming `viewIdea'` to `viewIdeaPage` or `viewIdeaData`?